src/nupic/engine/RegionImpl.cpp

```diff
@@ -230,7 +230,11 @@ void RegionImpl::setParameterArray(const std::string& name, Int64 index,const Ar
       break;
     }
 
-    NTA_ASSERT(rc == 0) << "getParameterArray - failure to get parameter '" << name << "' on node of type " << getType();
+    if (rc != 0)
```

Comment:
It's not clear why it was changed away from NTA_THROW. Here's the commit: https://github.com/numenta/nupic.core/commit/1212df77d9cbdb042b2a5e863951b3376f8f3593

This makes it consistent with the rest of the file.